mailbox/memory/src/test/java/org/apache/james/mailbox/inmemory/manager/InMemoryIntegrationResources.java

```diff
@@ -107,6 +113,116 @@ public InMemoryPerUserMaxQuotaManager getMaxQuotaManager() {
+        private Optional<Authenticator> authenticator;
+        private Optional<Authorizator> authorizator;
+        private Optional<EventBus> eventBus;
+        private Optional<Integer> limitAnnotationCount;
+        private Optional<Integer> limitAnnotationSize;
+        private ImmutableSet.Builder<BiFunction<SessionProvider, InMemoryMailboxSessionMapperFactory, PreDeletionHook>> preDeletionHooksInstanciators;
+
+        public Factory() {
+            this.groupMembershipResolver = Optional.empty();
+            this.authenticator = Optional.empty();
+            this.authorizator = Optional.empty();
+            this.eventBus = Optional.empty();
+            this.limitAnnotationCount = Optional.empty();
+            this.limitAnnotationSize = Optional.empty();
+            this.preDeletionHooksInstanciators = ImmutableSet.builder();
+        }
+
+        public Factory withAuthenticator(Authenticator authenticator) {
+            this.authenticator = Optional.of(authenticator);
+            return this;
+        }
+
+        public Factory withAuthorizator(Authorizator authorizator) {
+            this.authorizator = Optional.of(authorizator);
+            return this;
+        }
+
+        public Factory withEventBus(EventBus eventBus) {
+            this.eventBus = Optional.of(eventBus);
+            return this;
+        }
+
+        public Factory withGroupmembershipResolver(GroupMembershipResolver groupmembershipResolver) {
+            this.groupMembershipResolver = Optional.of(groupmembershipResolver);
+            return this;
+        }
+
+        public Factory withAnnotationLimits(int limitAnnotationCount, int limitAnnotationSize) {
+            this.limitAnnotationCount = Optional.of(limitAnnotationCount);
+            this.limitAnnotationSize = Optional.of(limitAnnotationSize);
+            return this;
+        }
+
+        public Factory withPreDeletionHooks(Collection<PreDeletionHook> preDeletionHooks) {
+            this.preDeletionHooksInstanciators.addAll(preDeletionHooks.stream()
+                .map(this::toInstanciator)
+                .collect(Guavate.toImmutableList()));
+            return this;
+        }
+
+        public Factory withPreDeletionHook(BiFunction<SessionProvider, InMemoryMailboxSessionMapperFactory, PreDeletionHook> preDeletionHook) {
+            this.preDeletionHooksInstanciators.add(preDeletionHook);
+            return this;
+        }
+
+        private BiFunction<SessionProvider, InMemoryMailboxSessionMapperFactory, PreDeletionHook> toInstanciator(PreDeletionHook preDeletionHook) {
+            return (a, b) -> preDeletionHook;
+        }
+
+        public Resources create() {
+            InMemoryMailboxSessionMapperFactory mailboxSessionMapperFactory = new InMemoryMailboxSessionMapperFactory();
+            EventBus eventBus = this.eventBus.orElseGet(() -> new InVMEventBus(new InVmEventDelivery(new NoopMetricFactory())));
```

Comment:
isn't it dangerous to have default implementations for a test env? I mean, you can very well forget to provide an EventBus and have green tests when you think you were testing Rabbit EventBus.
What about making a staged builder with Enum values to make the pain as small as possible while still be descriptive about we run?

Reply:
> isn't it dangerous to have default implementations for a test env?

IMO it is safe to expect from an `InMemoryIntegrationResources` to be based on an `InVMEventBus`

I don't see why this step would be special here. Why not these alays ask for the search index in the builder? After all we are not sure this is the scanning one. Why not ask for the explicit list of preDeletionHooks? How about the AnnotationLimits? 

If it's to duplicate everywhere:

````
new Factory()
   .preProvisionnedFakeAuthenticator()
   .fakeAuthorizator()
   .inVmEventBus()
   .defaultAnnotationLimits()
   .storeQuotaManager()
   .defaultMessageParser()
   .scanningSearchIndex()
   .noPreDeletionHooks()
   .create();
```

Isn't that a bit long?Are we sure we do want this?

Reply:
I did contribute that staged factory.

A convenient `public static InMemoryIntegrationResources defaultResources()` makes the above proposal reasonable.